include/sddf/block/blk_shared_queue.h

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright 2023, UNSW
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include "fence.h"
+
+/* Number of buffers the request queue is configured to have. */
+#define BLK_REQ_QUEUE_SIZE 1024
+/* Number of buffers the response queue is configured to have. */
+#define BLK_RESP_QUEUE_SIZE 1024
+
+typedef struct blk_storage_info {
+    char serial_number[64];
+    bool read_only;
+    bool ready;
+    uint16_t blocksize;
+    uint16_t queue_depth;
+    uint16_t cylinders, heads, blocks; /* geometry to guide FS layout */
+    uint64_t size; /* number of blocksize units */
+} blk_storage_info_t;
+
+/* Request code for block */
+typedef enum blk_request_code {
```

Comment:
why do we need to define request code in the library in stead of letting the users to decide what they need to have?

Reply:
the client + server will need to agree on some interface regardless. The idea of specifying the type of requests in the shared library is that they should cover all use cases for block devices

Reply:
This is part of the definition of the interface, it has to be here.